### ISSUE TITLE
E0025 needs to be updated to new format

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/rust#35198]
From: src/test/compile-fail/E0025.rs

E0025 needs span_label instead of span_note, changing it from:

```
error[E0025]: field `a` bound multiple times in the pattern
  --> src/test/compile-fail/E0025.rs:18:21
   |
18 |     let Foo { a: x, a: y, b: 0 } = x; //~ ERROR E0025
   |                     ^^^^
   |
note: field `a` previously bound here
  --> src/test/compile-fail/E0025.rs:18:15
   |
18 |     let Foo { a: x, a: y, b: 0 } = x; //~ ERROR E0025
   |               ^^^^
```

To:

```
error[E0025]: field `a` bound multiple times in the pattern
  --> src/test/compile-fail/E0025.rs:18:21
   |
18 |     let Foo { a: x, a: y, b: 0 } = x; //~ ERROR E0025
   |               ----  ^^^^ multiple uses of `a` in pattern
   |               |
   |               first use of `a`
```

**Comment by christopherdumas** [comment]
I'd like to handle this.

**Comment by jonathandturner** [comment]
@christopherdumas - thanks for volunteering!  Currently @pcn signed up for this on the [big error list](https://github.com/rust-lang/rust/issues/35233), though I'm thinking it's time for us to have more people working on the errors since some people get busy.

**Comment by christopherdumas** [comment]
I'll work on this, but if @pcn gets to it first, I'll not make a pull request.

**Comment by pcn** [comment]
This one is more complex than the others I've looked at. The current code in `_match.rs` looks like this:

https://github.com/rust-lang/rust/blob/8eb12d91aaf95432ca73bda429af04e0710c984d/src/librustc_typeck/check/_match.rs#L670-L680

Naively changing the span_note to a span_label seems to cause an issue and I digressed to read  more about  the borrow checker, etc.  but haven't been  able to translate  that into a good understanding and a fix. @christopherdumas if you've already had experience with this,  I'd love to learn from you.
